src/Serilog.Sinks.OpenTelemetry/Sinks/OpenTelemetry/Formatting/CleanMessageTemplateFormatter.cs

```diff
@@ -107,36 +107,9 @@ static void RenderPropertyValueUnaligned(LogEventPropertyValue propertyValue, Te
             return;
         }
 
-        if (value is ValueType)
+        if (formatProvider?.GetFormat(typeof(ICustomFormatter)) is ICustomFormatter customFormatter)
         {
-            if (value is int or uint or long or ulong or decimal or byte or sbyte or short or ushort)
-            {
-                output.Write(((IFormattable)value).ToString(format, formatProvider));
-                return;
-            }
-
-            if (value is double d)
-            {
-                output.Write(d.ToString(format, formatProvider));
-                return;
-            }
-
-            if (value is float f)
-            {
-                output.Write(f.ToString(format, formatProvider));
-                return;
-            }
-
-            if (value is bool b)
-            {
-                output.Write(b);
-                return;
-            }
-        }
-
-        if (value is IFormattable formattable)
-        {
-            output.Write(formattable.ToString(format, formatProvider));
```

Comment:
The deleted code here is all still required, without it, the format provider isn't passed through `IFormattable`.

Reply:
I don't think so. Custom formatters often do this internally if they see fit.

Reply:
If there's no custom formatter, though, we just fall through to `output.Write(value)` (no format provider used).